File: java-concert/src/main/java/prac/javaconcert/concert/controller/ConcertController.java
package prac.javaconcert.concert.controller;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import prac.javaconcert.concert.request.ConcertSaveRequest;
import prac.javaconcert.concert.response.ConcertDetailResponse;
import prac.javaconcert.concert.response.ConcertPreviewResponse;
import prac.javaconcert.concert.service.ConcertService;

import java.util.List;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/v1/concert")
public class ConcertController {

    private final ConcertService concertService;

    @GetMapping
    public List<ConcertPreviewResponse> findAll(){
        return concertService.findAll();
    }

    @PostMapping
    public ConcertDetailResponse save(@RequestBody ConcertSaveRequest concertSaveRequest){
        return concertService.save(concertSaveRequest);
    }
}

File: concert/src/main/kotlin/prac/concert/controller/ConcertController.kt
package prac.concert.controller

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*
import prac.concert.request.ConcertSaveRequest
import prac.concert.response.ConcertDetailResponse
import prac.concert.service.ConcertService

@RestController
@RequestMapping("/api/v1")
class ConcertController(
    private val concertService : ConcertService
){

//    @Autowired
//    lateinit var concertService : ConcertService

    @GetMapping("/concert")
    fun findAll() : List<ConcertDetailResponse> {
        return concertService.findAll();
    }

    @PostMapping("/concert")
    fun save(@RequestBody concertSaveRequest : ConcertSaveRequest) : ConcertDetailResponse{
        return concertService.save(concertSaveRequest);
    }
}